Weather App/Controller/HomeViewController.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  HomeViewController.swift
+//  Weather App
+//
+//  Created by Sudipta Chowdhury on 5/21/24.
+//
+
+import UIKit
+
+class HomeViewController: BaseViewController,
+                          UITableViewDelegate,
+                          UITableViewDataSource {
+    
+    private let viewModel = HomeViewModel()
+    
+    @IBOutlet weak var table: UITableView!
+    
+    var refreshControl = UIRefreshControl()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        table.register(TemperatureTableViewCell.nib(), forCellReuseIdentifier: TemperatureTableViewCell.cellName)
+        table.register(InformationTableViewCell.nib(), forCellReuseIdentifier: InformationTableViewCell.cellName)
+        
+        refreshControl.addTarget(self, action: #selector(pullToRefresh(sender:)), for: .valueChanged)
+        
+        table.delegate = self
+        table.dataSource = self
+        table.refreshControl = refreshControl
+        table.refreshControl?.tintColor = .white
+        
+        fetchData()
+    }
+    
+    @objc func pullToRefresh(sender: UIRefreshControl) {
+        fetchData()
+    }
+    
+    private func fetchData() {
+        viewModel.fetchCurrentWeatherInfo {
+            let error = self.viewModel.getError()
+            if error == nil {
+                self.updateUI()
+            } else {
+                print(error!)
+            }
+        }
+    }
+    
+    private func updateUI() {
+        DispatchQueue.main.async {
+            self.table.reloadData()
+            self.table.refreshControl?.endRefreshing()
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 2
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        return showTableCell(tableView: tableView, indexPath: indexPath)
+    }
+    
+    func showTableCell(tableView: UITableView, indexPath: IndexPath) -> UITableViewCell {
+        switch (indexPath.row) {
+        case 0:
```

Comment:
Please create an enum for the cells.
Case 0/1 doesn't express anything about the cells